### PR TITLE
[DependencyInjection] force enabling the external XML entity loaders

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18876, #18908 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by xabbuh** [comment]
@nicolas-grekas @sstok I don't see the security issue here. The `XmlUltils` class still disables the entity loaders and switches the setting back to the old value after documents have been processed. The only place where we did not take into account whether the entity loaders are disabled are the validation of DI extension config files (which currently implicitly assumes that loaders are not disabled).

**Comment by sstok** [comment]
What I mean is will `schemaValidateSource()` load any external resources of the source or will it parse the source as-is? And if it will load external resources, are then any security risks with that (including DoS attacks).

**Comment by fabpot** [comment]
There are other calls to `schemaValidateSource` in some other components, do we also need to make the same change?

**Comment by xabbuh** [comment]
@fabpot One place is in the XliffFileLoader which afaik doesn't support to import resources anyway (and thus would not be affected). And forcing the value in the `XmlUtils` does not sound like a good idea to me as we do not know how people use that class and imo they should force the proper value themselves if necessary.

**Comment by xabbuh** [comment]
Ah no I was mistaken. The `XliffFileLoader` needs to be updated as well.

File: src/Symfony/Component/Translation/Loader/XliffFileLoader.php [hunk 1]
```diff
@@ -143,10 +143,14 @@ private function parseFile($file)
         $source = file_get_contents(__DIR__.'/schema/dic/xliff-core/xliff-core-1.2-strict.xsd');
         $source = str_replace('http://www.w3.org/2001/xml.xsd', $location, $source);
 
+        $disableEntities = libxml_disable_entity_loader(false);
+
         if (!@$dom->schemaValidateSource($source)) {
             throw new InvalidResourceException(sprintf('Invalid resource provided: "%s"; Errors: %s', $file, implode("\n", $this->getXmlErrors($internalErrors))));
         }
 
+        libxml_disable_entity_loader($disableEntities);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Translation/Loader/XliffFileLoader.php]: we need to restore the orig value before throwing, isn't it?

Comment (xabbuh) [inline src/Symfony/Component/Translation/Loader/XliffFileLoader.php]: Sure, good catch.

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**